src/Symfony/Component/Notifier/Bridge/OvhCloud/OvhCloudTransport.php

```diff
@@ -0,0 +1,138 @@
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Thomas Ferney <thomas.ferney@gmail.com>
+ *
+ * @experimental in 5.1
+ */
+final class OvhCloudTransport extends AbstractTransport
+{
+    private $endpoints = [
+        'ovh-eu' => 'https://eu.api.ovh.com/1.0',
+        'ovh-ca' => 'https://ca.api.ovh.com/1.0',
+        'ovh-us' => 'https://api.us.ovhcloud.com/1.0',
+        'kimsufi-eu' => 'https://eu.api.kimsufi.com/1.0',
+        'kimsufi-ca' => 'https://ca.api.kimsufi.com/1.0',
+        'soyoustart-eu' => 'https://eu.api.soyoustart.com/1.0',
+        'soyoustart-ca' => 'https://ca.api.soyoustart.com/1.0',
+        'runabove-ca' => 'https://api.runabove.com/1.0',
+    ];
+
+    private $applicationKey;
+    private $applicationSecret;
+    private $consumerKey;
+    private $serviceName;
+    private $timeDelta;
+
+    public function __construct(string $applicationKey, string $applicationSecret, string $consumerKey, string $serviceName, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->applicationKey = $applicationKey;
+        $this->applicationSecret = $applicationSecret;
+        $this->consumerKey = $consumerKey;
+        $this->serviceName = $serviceName;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function setHostByEndpoint(?string $endpoint): self
+    {
+        $this->host = $this->endpoints[$endpoint] ?: self::HOST;
+
+        return $this;
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('ovhcloud://%s?consumer_key=%s&service_name=%s', $this->getEndpoint(), $this->consumerKey, $this->serviceName);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): void
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, SmsMessage::class, \get_class($message)));
+        }
+
+        $endpoint = sprintf('%s/sms/%s/jobs', $this->getEndpoint(), $this->serviceName);
+
+        $content = [
+            'charset' => 'UTF-8',
+            'class' => 'flash',
+            'coding' => '8bit',
+            'message' => $message->getSubject(),
+            'receivers' => [$message->getPhone()],
+            'noStopClause' => false,
+            'priority' => 'medium',
+            'senderForResponse' => true,
+        ];
+
+        $headers['X-Ovh-Application'] = $this->applicationKey;
+
+        if (!isset($this->timeDelta)) {
+            $this->calculateTimeDelta();
+        }
+
+        $now = time() + $this->timeDelta;
+        $headers['X-Ovh-Timestamp'] = $now;
+
+        if (isset($this->consumerKey)) {
```

Comment:
`if (null === $this->consumerKey) {`

Reply:
If my assumption is right above, the condition must be removed and local variables should be used instead.

Reply:
The "consumerKey" is mandatory, so I think I can remove this condition.

If it's empty, it throws an TransportException with "Missing X-Ovh-Consumer header"
If it's undefined in the dsn, it throws an 500 with null instead of a string in __construct()

What do you think?

Reply:
indeed, that's always set, so you can remove the condition.